File: backend/src/main/java/com/example/ubernet/dto/DriverDto.java
package com.example.ubernet.dto;

import com.example.ubernet.model.DriverDailyActivity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DriverDto {
    private String email;
    private String password;
    private String name;
    private String surname;
    private String city;
    private String phoneNumber;
    public boolean isWorking;
    private boolean blocked;
    private DriverDailyActivity driverDailyActivity;
    private boolean requestedChanges;
}

File: frontend/src/app/model/DriverDTO.ts
export class DriverDTO {
  email!: string;
  name!: string;
  surname!: string;
  city!: string;
  phoneNumber!: string;
  role!: string;
  isWorking: boolean;
  requestedChanges: boolean;

  constructor(name: string, surname: string, email: string, phoneNumber: string, city: string, isWorking: boolean) {
    this.name = name;
    this.surname = surname;
    this.email = email;
    this.city = city;
    this.phoneNumber = phoneNumber;
    this.isWorking = isWorking;
  }
}
